benchmarks/bench_ml.py

```diff
@@ -0,0 +1,422 @@
+            70% of data is used for training, rest for validation.
+    """
+
+    X, Xv, y, yv = train_test_split(X, y, train_size=0.7)
+
+    sc = StandardScaler() # normalize data
+    sc.fit(X, y)
+
+    X, Xv = sc.transform(X), sc.transform(Xv)
+
+    return X, y, Xv, yv
+
+# functions below are used to apply non - linear maps to parameter values, eg
+# -3.0 -> 0.001
+
+def pow10map(x):
+    return 10.0 ** x
+
+def pow2intmap(x):
+    return int(2.0 ** x)
+
+def nop(x):
+    return x
+
+# this is used to process the output of fetch_mldata
+def load_data_target(data):
+    X = data['data']
+    Y = data['target']
+    return X, Y
+
+class TesterClass():
+    """
+    A class which is used to perform benchmarking of black box optimization algorithms on various
+    machine learning problems.
+    On __init__, the dataset is loaded that is used for experimentation, and it is kept in memory
+    in order to avoid reloading data.
+    """
+    nnparams = {
+        # up to 1024 neurons
+        'hidden_layer_sizes': (Real(1.0, 10.0), pow2intmap),
+        'activation': (Categorical(['identity', 'logistic', 'tanh', 'relu']), nop),
+        'solver': (Categorical(['lbfgs', 'sgd', 'adam']), nop),
+        'alpha': (Real(-5.0, -1), pow10map),
+        'batch_size': (Real(5.0, 10.0), pow2intmap),
+        'learning_rate': (Categorical(['constant', 'invscaling', 'adaptive']), nop),
+        'max_iter': (Real(5.0, 8.0), pow2intmap),
+        'learning_rate_init': (Real(-5.0, -1), pow10map),
+        'power_t': (Real(0.01, 0.99), nop),
+        'momentum': (Real(0.1, 0.98), nop),
+        'nesterovs_momentum': (Categorical([True, False]), nop),
+        'beta_1': (Real(0.1, 0.98), nop),
+        'beta_2': (Real(0.1, 0.9999999), nop),
+    }
+
+    # every model should have fields:
+    # OUTPUT_TYPE:  OUTPUT_CONTINUOUS / OUTPUT_CATEGORY
+    # MODEL_PARAMETERS: dictionary whose keys correspond to parameters that are passed to __init__ of python class
+    #                   that implements model
+    # MODEL_BACKEND: python class that implements model
+    models = {
+        #
+        #   regressors go here
+        #
+        MLPRegressor.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: nnparams,
+            MODEL_BACKEND: MLPRegressor,
+        },
+        SVR.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: {
+                'C': (Real(-4.0, 4.0), pow10map),
+                'epsilon': (Real(-4.0, 1.0), pow10map),
+                'gamma': (Real(-4.0, 1.0), pow10map),
+            },
+            MODEL_BACKEND: SVR,
+        },
+        DecisionTreeRegressor.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: {
+                'max_depth': (Real(1.0, 4.0), pow2intmap),
+                'min_samples_split': (Real(1.0, 8.0), pow2intmap),
+            },
+            MODEL_BACKEND: DecisionTreeRegressor,
```

Comment:
If you make this an instance instead of a class, you can infer `OUTPUT_TYPE` using `isinstance(model, RegressorMixin)` or `isinstance(model, ClassifierMixin)` where the mixins are inherited from `sklearn.base`